### PR TITLE
[DOCs ADD] Add doc about how to build customer library to encapsulate openvino

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
Add doc about how to build customer library to encapsulate openvino

Signed-off-by: xuejun <Xuejun.Zhai@intel.com>

### Details:
 - A doc about how to build customer library to encapsulate openvino
 - A simple example
 - A simple test case to verified the built library

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-86999

### CODE REVIEW

File: docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+$ cd library-example
+$ mkdir build && cd build
+$ cmake ..
+```
+
+* build to generate the libcustomer-library.so
+```sh
+$ make
+$ ll | grep lib
+-rwxrwxr-x 1 odt odt 39744  七  28 13:28 libcustomer-library.so*
```
Comment (xipingyan) [inline docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md:71]: Chinese character？ “七”

Comment (zhaixuejun1993) [reply docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md]: Done

File: docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+* compiler the sample
+```sh
+g++ -o mytest main.cpp -L. -lcustomer-library
+```
+
+* run the sample
+```sh
+./TEST
```
[the remark this answers is not in the document]
Comment (zhaixuejun1993) [reply docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md:97]: Done

File: docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# How to build library to encapsulate OpenVINO
+
+> **NOTE**: This guide illustrated a simplified example about how to build a customer library based on OpenVINO.
```
Comment (ilya-lavrenov) [inline docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md:3]: why is it needed? We have https://docs.openvino.ai/latest/openvino_docs_OV_UG_Integrate_OV_with_your_application.html#link-and-build-your-c-application-with-openvino-runtime

Comment (riverlijunjie) [reply docs/HOWTO/how_to_build_library_to_encapsulate_openvino.md:3]: A difference is one is to build into application and one is to build into a shared library.  Though they are only tiny different, but some customer don't know how to do it. So we should add this part.
Anyhow, it should be better to put this part into https://docs.openvino.ai/latest/openvino_docs_OV_UG_Integrate_OV_with_your_application.html#link-and-build-your-c-application-with-openvino-runtime ?

**Comment by riverlijunjie** [comment]
Closed due to it is not openvino scope.

**Pull request closed without merging**